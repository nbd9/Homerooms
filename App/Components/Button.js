import React from 'react'
import { TouchableOpacity, Text, StyleSheet, View } from 'react-native'
import { Fonts, Colors } from '../../Extras/Themes'

const Button = ({text, onPress, disabled = false, children}) => {
  return (
    <TouchableOpacity style={(disabled) ? styles.disabledButton : styles.button} onPress={(disabled) ? null : onPress} >
      {
        (children != null)
          ? children
          : <Text style={styles.buttonText}>{text}</Text>
      }
    </TouchableOpacity>
  )
}

const styles = StyleSheet.create({
  button: {
    backgroundColor: Colors.darkBlue,
    borderRadius: 24,
    paddingBottom: 10,
    paddingLeft: 50,
    paddingRight: 50
    // flex: 1,
    // flexDirection: 'row',
    // justifyContent: 'center'
  },
  buttonText: {
    fontFamily: Fonts.type.content,
    paddingTop: 10,
    fontSize: 24,
    color: 'white',
    textAlign: 'center'
  },
  disabledButton: {
    backgroundColor: Colors.disabled,
    borderRadius: 24,
    paddingBottom: 10,
    paddingLeft: 50,
    paddingRight: 50
    // flex: 1,
    // flexDirection: 'row',
    // justifyContent: 'center'
  }
})

export default Button
