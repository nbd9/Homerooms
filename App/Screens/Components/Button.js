import React from 'react'
import {
  TouchableOpacity,
  Text,
  StyleSheet
} from 'react-native'
import { Fonts, Colors } from '../../Themes'

const Button = ({text, onPress}) => {
  return (
    <TouchableOpacity style={styles.button} onPress={onPress} >
      <Text style={styles.buttonText}>{text}</Text>
    </TouchableOpacity>
  )
}

const styles = StyleSheet.create({
  button: {
    backgroundColor: Colors.darkBlue,
    borderRadius: 24,
    paddingBottom: 10,
    paddingLeft: 50,
    paddingRight: 50
  },
  buttonText: {
    fontFamily: Fonts.type.content,
    paddingTop: 10,
    fontSize: 24,
    color: 'white',
    textAlign: 'center'
  }
})

export default Button
