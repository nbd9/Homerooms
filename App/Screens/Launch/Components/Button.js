import React from 'react'
import {
  View,
  Text,
  StyleSheet
} from 'react-native'

const Button = ({text: buttonText}) => {
  return (
    <View style={styles.button}>
      <Text style={styles.buttonText}>{buttonText}</Text>
    </View>
  )
}

const styles = StyleSheet.create({
  button: {
    backgroundColor: '#3B5998',
    borderRadius: 24,
    paddingBottom: 10,
    paddingLeft: 50,
    paddingRight: 50
  },
  buttonText: {
    fontFamily: 'Lato-Bold',
    paddingTop: 10,
    fontSize: 24,
    color: '#FFFFFF',
    textAlign: 'center'
  }
})

export default Button
