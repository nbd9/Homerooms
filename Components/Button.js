import React from 'react'
import { TouchableOpacity, Text, StyleSheet } from 'react-native'
import { Fonts, Colors } from '../Themes'

const Button = ({text, onPress, disabled = false, children}) => {
  return (
    <TouchableOpacity style={(disabled) ? styles.disabledButton : styles.button} onPress={(disabled) ? null : onPress} >
      {
        (children != null)
          ? children
          : <Text style={styles.buttonText}>{text}</Text>
      }
    </TouchableOpacity>
  )
}

const styles = StyleSheet.create({
  button: {
    backgroundColor: Colors.darkBlue,
    borderRadius: 24,
    paddingBottom: 10,
    paddingLeft: 50,
    paddingRight: 50
  },
  buttonText: {
    fontFamily: Fonts.type.content,
    paddingTop: 10,
    fontSize: 24,
    color: 'white',
    textAlign: 'center'
  },
  disabledButton: {
    backgroundColor: Colors.disabled,
    borderRadius: 24,
    paddingBottom: 10,
    paddingLeft: 50,
    paddingRight: 50
  }
})

export default Button
