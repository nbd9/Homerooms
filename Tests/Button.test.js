import 'react-native'
import React from 'react'
import Button from '../App/Screens/Components/Button'

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer'

test('renders correctly', () => {
  const tree = renderer.create(<Button text='Testing' />).toJSON()
  expect(tree).toMatchSnapshot()
})
