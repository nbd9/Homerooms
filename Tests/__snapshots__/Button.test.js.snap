// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#3B5998",
      "borderRadius": 24,
      "opacity": 1,
      "paddingBottom": 10,
      "paddingLeft": 50,
      "paddingRight": 50,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "white",
        "fontFamily": "Nunito",
        "fontSize": 24,
        "paddingTop": 10,
        "textAlign": "center",
      }
    }
  >
    Testing
  </Text>
</View>
`;
